src/java/picard/illumina/ClusterDataToSamConverter.java

```diff
@@ -101,10 +107,39 @@ public ClusterDataToSamConverter(final String runBarcode,
     }
 
     /**
+     *  Sets the SAM tag to use to store the molecular index bases.  If multiple molecular indexes exist, it will concatenate them
+     * and store them in this tag.
+     */
+    public ClusterDataToSamConverter withMolecularIndexTag(final String molecularIndexTag) {
+        this.molecularIndexTag = molecularIndexTag;
```

Comment:
Actually, now that you mention empty strings....

null -> error.
empty string -> no tag. 

would be my preference.